gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/inmemory/SharedPolicyGroupCrudServiceInMemory.java

```diff
@@ -72,6 +81,25 @@ public Optional<SharedPolicyGroup> findByEnvironmentIdAndCrossId(String environm
             .findFirst();
     }
 
+    @Override
+    public Optional<SharedPolicyGroup> getLastDeployedByEnvironmentIdAndCrossId(String environmentId, String crossId) {
+        var sharedPolicyGroupOpt = findByEnvironmentIdAndCrossId(environmentId, crossId);
+        if (sharedPolicyGroupOpt.isPresent()) {
+            var sharedPolicyGroup = sharedPolicyGroupOpt.get();
+            if (sharedPolicyGroup.isDeployed()) {
+                return sharedPolicyGroupOpt;
+            }
+            var historySharedPolicyGroupOpt = sharedPolicyGroupHistoryQueryService.getLatestBySharedPolicyGroupId(
```

Comment:
It seems strange to me that an InMemory service needs to call a QueryService. 

I don't think this is necessary because it complicates testing and you end up with a mock. And this is just to test code in inMemory, which is not intended for production.